virttest/qemu_vm.py

```diff
@@ -4473,7 +4529,33 @@ def change_media(self, device, target):
         :param device: Device ID;
         :param target: new media file;
         """
-        return self.monitor.change_media(device, target)
+        status_before = status_after = self.check_device_tray_opened(device)
+
+        qmp_monitors = [m for m in self.monitors if m.protocol == "qmp"]
+        for m in qmp_monitors:
+            m.clear_events()
+
+        out = self.monitor.change_media(device, target)
+
+        for m in qmp_monitors:
+            events = utils_misc.wait_for(m.get_events(), timeout=120)
+            logging.info("Event list:\n%s" % str(events))
+            count = 0
+            for event in events:
+                if event["event"] == u"DEVICE_TRAY_MOVED":
+                    count += 1
+                    status_after = bool(event["data"]["tray-open"])
```

Comment:
why not call check_device_tray_opened() again here?

Reply:
no need, since check the events received could get the final status.